crates/oxc_ast/src/serialize.rs

```diff
@@ -505,6 +505,75 @@ impl ESTree for ExportAllDeclarationWithClause<'_, '_> {
     }
 }
 
+#[ast_meta]
+#[estree(raw_deser = "
+        const start = DESER[u32](POS_OFFSET.span.start);
+        const end = DESER[u32](POS_OFFSET.span.end);
+        const attributesKeyword = DESER[IdentifierName](POS_OFFSET.attributes_keyword);
+        const properties = DESER[Vec<TSImportAttribute>](POS_OFFSET.elements);
+        const result = [
+            {
+                type: 'Property',
+                start,
+                end,
+                method: false,
+                shorthand: false,
+                computed: false,
+                key: attributesKeyword,
+                value: {
+                    type: 'ObjectExpression',
+                    start,
+                    end,
+                    properties,
+                },
+                kind: 'init',
+            },
+        ];
+        result
+    ")]
+pub struct TSImportAttributesProperties<'a, 'b>(pub &'b TSImportAttributes<'a>);
+
+impl ESTree for TSImportAttributesProperties<'_, '_> {
+    fn serialize<S: Serializer>(&self, serializer: S) {
+        let mut state = serializer.serialize_sequence();
+        state.serialize_element(&TSImportAttributesPropertiesWithProperty(self.0));
+        state.end();
+    }
+}
+
+pub struct TSImportAttributesPropertiesWithProperty<'a, 'b>(pub &'b TSImportAttributes<'a>);
+
+impl ESTree for TSImportAttributesPropertiesWithProperty<'_, '_> {
+    fn serialize<S: Serializer>(&self, serializer: S) {
+        let mut state = serializer.serialize_struct();
+        state.serialize_field("type", &JsonSafeString("Property"));
+        // TODO: span
+        state.serialize_field("start", &self.0.span.end);
+        state.serialize_field("end", &self.0.span.end);
+        state.serialize_field("method", &false);
+        state.serialize_field("shorthand", &false);
+        state.serialize_field("computed", &false);
+        state.serialize_field("key", &self.0.attributes_keyword);
+        state.serialize_field("value", &TSImportAttributesPropertiesWithPropertyValue(self.0));
+        state.serialize_field("kind", &"init");
+        state.end();
+    }
+}
+
+struct TSImportAttributesPropertiesWithPropertyValue<'a, 'b>(&'b TSImportAttributes<'a>);
+
+impl ESTree for TSImportAttributesPropertiesWithPropertyValue<'_, '_> {
+    fn serialize<S: Serializer>(&self, serializer: S) {
+        let mut state = serializer.serialize_struct();
+        state.serialize_field("type", &JsonSafeString("ObjectExpression"));
+        // TODO: span
+        state.serialize_field("start", &self.0.span.start);
+        state.serialize_field("end", &self.0.span.end);
+        state.serialize_field("properties", &self.0.elements);
```

Comment:
As commented in https://github.com/oxc-project/oxc/issues/9663#issuecomment-2719823864, there's no span information for inner braces to serialize this as `ObjectExpression`. 

```ts
type A = import("foo", {  with:  {  foo: "foo"  }  })
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ <-- TSImportAttributes.span
                          ^^^^                       <-- TSImportAttributes.attributes_keyword.span
                                 ^^^^^^^^^^^^^^^^    <-- ???
                                    ^^^^^^^^^^       <-- TSImportAttribute.span
```